cmd/webhook-manager/app/util.go

```diff
@@ -17,93 +17,96 @@ limitations under the License.
+		"volcano-admission-service-queues-mutate",
+		"volcano-admission-service-podgroups-mutate",
+		"volcano-admission-service-jobs-mutate",
 	}
-	if config.IgnoredNamespaces != "" {
-		ignoredNamespaces := strings.Split(strings.TrimSpace(config.IgnoredNamespaces), ",")
-		klog.Infof("The ignored namespaces list of webhook manager is <%v>.",
-			ignoredNamespaces)
-		webhookLabelSelector = &metav1.LabelSelector{
-			MatchExpressions: []metav1.LabelSelectorRequirement{
-				{
-					Values:   ignoredNamespaces,
-					Operator: "NotIn",
-					Key:      "kubernetes.io/metadata.name",
-				},
-			},
-		}
-	}
-	if service.MutatingConfig != nil {
-		for i := range service.MutatingConfig.Webhooks {
-			service.MutatingConfig.Webhooks[i].SideEffects = &sideEffect
-			service.MutatingConfig.Webhooks[i].AdmissionReviewVersions = reviewVersions
-			service.MutatingConfig.Webhooks[i].ClientConfig = clientConfig
-			service.MutatingConfig.Webhooks[i].NamespaceSelector = webhookLabelSelector
-		}
+)
 
-		service.MutatingConfig.ObjectMeta.Name = webhookConfigName(config.WebhookName, service.Path)
+func addCaCertForWebhook(kubeClient *kubernetes.Clientset, caBundle []byte) error {
+	for _, mutatingWebhookName := range mutatingWebhooksName {
+		var mutatingWebhook *v1.MutatingWebhookConfiguration
+		if err := wait.Poll(time.Second, 5*time.Minute, func() (done bool, err error) {
+			mutatingWebhook, err = kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.TODO(), mutatingWebhookName, metav1.GetOptions{})
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					klog.Errorln(err)
+					return false, nil
+				}
+				return false, fmt.Errorf("failed to get mutating webhook %v", err)
+			}
+			return true, nil
+		}); err != nil {
+			return fmt.Errorf("failed to get mutating webhook %v", err)
+		}
 
-		if err := registerMutateWebhook(kubeClient, service.MutatingConfig); err != nil {
-			klog.Errorf("Failed to register mutating admission webhook (%s): %v",
-				service.Path, err)
-		} else {
-			klog.V(3).Infof("Registered mutating webhook for path <%s>.", service.Path)
+		for index := 0; index < len(mutatingWebhook.Webhooks); index++ {
+			if mutatingWebhook.Webhooks[index].ClientConfig.CABundle == nil ||
+				bytes.Equal(mutatingWebhook.Webhooks[index].ClientConfig.CABundle, caBundle) {
+				mutatingWebhook.Webhooks[index].ClientConfig.CABundle = caBundle
+			}
 		}
-	}
-	if service.ValidatingConfig != nil {
-		for i := range service.ValidatingConfig.Webhooks {
-			service.ValidatingConfig.Webhooks[i].SideEffects = &sideEffect
-			service.ValidatingConfig.Webhooks[i].AdmissionReviewVersions = reviewVersions
-			service.ValidatingConfig.Webhooks[i].ClientConfig = clientConfig
-			service.ValidatingConfig.Webhooks[i].NamespaceSelector = webhookLabelSelector
+		if _, err := kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Update(context.TODO(), mutatingWebhook, metav1.UpdateOptions{}); err != nil {
```

Comment:
it's better to compare the updated webhook with original mutatingWebhook here, if no change, we don't need to call Update func

Reply:
Ok, I added a flag to record if the webhook has been changed, and if so. then call the update function. reflection is not used, which may affect performance